src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -637,6 +639,264 @@ def _decode_inhibit_any_policy(backend, asn1_int):
     return x509.InhibitAnyPolicy(skip_certs)
 
 
+@utils.register_interface(x509.RevokedCertificate)
+class _RevokedCertificate(object):
+    def __init__(self, backend, x509_revoked):
+        self._backend = backend
+        self._x509_revoked = x509_revoked
+
+        self._serial_number = None
+        self._revocation_date = None
+        self._extensions = None
+
+    @property
+    def serial_number(self):
+        if self._serial_number:
+            return self._serial_number
+
+        asn1_int = self._x509_revoked.serialNumber
+        self._backend.openssl_assert(asn1_int != self._backend._ffi.NULL)
+        self._serial_number = self._backend._asn1_integer_to_int(asn1_int)
+        return self._serial_number
+
+    @property
+    def revocation_date(self):
+        if self._revocation_date:
+            return self._revocation_date
+
+        self._revocation_date = self._backend._parse_asn1_time(
+            self._x509_revoked.revocationDate)
+        return self._revocation_date
+
+    @property
+    def extensions(self):
+        if self._extensions:
+            return self._extensions
+
+        extensions = []
+        seen_oids = set()
+        extcount = self._backend._lib.X509_REVOKED_get_ext_count(
+            self._x509_revoked)
+        for i in range(extcount):
+            ext = self._backend._lib.X509_REVOKED_get_ext(
+                self._x509_revoked, i)
```

Comment:
Can this code be written in terms of the extension parser class, it looks like yet?